File: TP01/ExerciciosProva/PapaiNoel.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char* carta(char* nome1, char* nome2){
    char* cartinha = (char*)malloc(sizeof(char) * 40);

    int tamTotal = strlen(nome1) + strlen(nome2);

    for(int i = 0, j = 0, z = 0; i < tamTotal; i += 4, j+=2, z+=2){
         cartinha[i] = nome1[j];
         cartinha[i + 1] = nome1[j + 1];
         cartinha[i + 2] = nome2[z];
         cartinha[i + 3] = nome2[z + 1];
    }

    return cartinha;
}

int main(){
    int casos;
    scanf("%d", &casos);

    for(int i = 0; i < casos; i++){
       char nome1[20];
       char nome2[20];
       scanf(" %[^\n]", nome1);
       scanf(" %[^\n]", nome2);
       char* cartinha = carta(nome1, nome2);
       printf("%s\n" , cartinha);
       free(cartinha);
    }
}

File: TP01/ExerciciosProva/PapaiNoel.java
import java.util.*;

public class App{

    public static String cartinha(String nome1, String nome2){
        int tam = nome1.length() + nome2.length() - 1;
        String nome3 = "";
        int j = 0, z = 0, temp = 0;

        for(int i = 0; i < tam; i +=2, temp += 1){
            if(temp % 2 == 0){
               char a = nome1.charAt(j);
               nome3 += a;
               char b = nome1.charAt(j+1);
               nome3 += b;
               j +=2;
            }else{
                char c = nome2.charAt(z);
                nome3 += c;
                char d = nome2.charAt(z + 1);
                nome3 += d;
                z += 2;
            }
        }

        return nome3.toString();
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in); 
        int casos = scanner.nextInt();
        scanner.nextLine();

        for(int i = 0; i < casos; i++){
            String nome1 = scanner.nextLine();
            String nome2 = scanner.nextLine();
            System.out.println(cartinha(nome1, nome2));
        }
       scanner.close();
    }
}